packages/hardhat-chai-matchers/src/internal/emit.ts

```diff
@@ -205,20 +230,21 @@ const tryAssertArgsArraysEqual = (
           Assertion,
           chaiUtils,
           expectedArgs,
-          logs[index]
+          logs[index],
+          assert,
+          ssfi
         );
         return;
       } catch {}
     }
   }
   const eventName = chaiUtils.flag(context, "eventName");
-  context.assert(
+  assert(
     false,
     `The specified arguments (${util.inspect(
       expectedArgs
     )}) were not included in any of the ${
       context.logs.length
-    } emitted "${eventName}" events`,
-    "Do not combine .not. with .withArgs()"
```

Comment:
Did we lose this warning message entirely?

Reply:
I removed it because it was unreachable (that's why no test failed after I did that).

Probably a better option would've been to have a separate method like `assertFalse` that just accepts a single argument, but that would've also made `buildAssert` return multiple functions or something like that...

But now that I think about it, I think the best option is to assert in the assert method (yeah, I know) that the message is defined for the condition that matches. That is: throw an error if the condition is true, negated is true, and the message for that combination does not exist. I will do that.

Reply:
Done in 7c2ebf745f93487c50167fc710dd1d998e21dced, please let me know what you think.